src/Bridge/Symfony/Bundle/DependencyInjection/Configuration.php

```diff
@@ -49,6 +49,7 @@ public function getConfigTreeBuilder()
                         ->booleanNode('enabled')->defaultTrue()->info('To enable or disable eager loading')->end()
                         ->integerNode('max_joins')->defaultValue(30)->info('Max number of joined relations before EagerLoading throws a RuntimeException')->end()
                         ->booleanNode('force_eager')->defaultTrue()->info('Force join on every relation. If disabled, it will only join relations having the EAGER fetch mode.')->end()
+                        ->booleanNode('filter_relations')->defaultFalse()->info('Allow to filter the relation collection when the filter is on a nested property. By default, it will retrieve the full relation of the filtered resource. If enabled, only relations matching the filter will be returned.')->end()
```

Comment:
Yeah your right I think that this option is too much confusing, ManyToMany relations are working fine, I've added a test case anyway. 

I'm removing the availability to filter on the relation. I think that the best to filter a relation would be to use `subresource` + filter.